sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -52,6 +54,9 @@ public final class CoreUtils {
 
     private static final char[] LOWERCASE_HEX_CHARACTERS = "0123456789abcdef".toCharArray();
 
+    // Used to check if the ISO8601 date time doesn't have a colon in the offset.
+    private static final Pattern ISO8601_COLONLESS_OFFSET = Pattern.compile("([+-][0-9]{2})([0-9]{2})(?=\\[|$)");
```

Comment:
Part of the time would contain `-` e.g. `2023-09-26`, but I assume time part won't match `-\d\d\d\d`, at least not in rfc3339.

Reply:
The issue this is fixing arises when the datetime string has an offset without a colon (+/-dddd), so the overall format is something like dddd-dd-ddTdd:dd:dd(+/-)dddd with an optional offset locale ([locale]). This shouldn't inadvertently match any other portion of the datetime string, except in a case where it's already invalid and non-parseable.

But ideally, this would have been resolvable by having a `DateTimeFormatter` that supported both `+/-dd:dd` and `+/-dddd` offset formats at the same time.